app/controllers/submissions_controller.rb

```diff
@@ -1,5 +1,50 @@
 class SubmissionsController < ApplicationController
   def index
+    # @user = current_user
+    # @submission = Submission.new
+
+    #code from index
+    @goal = Goal.find(params["goal_id"])
     @submissions = Submission.all
+    @submission = @submissions.first
+
+    # code from create
+
+
+    # Its going to need the goal instance in order for it to submit for the correct goal
+    # Find the id of the goal itself??? Can't be params[:id] cause its not exactly from a form. Its form a previous form
+    # @submission.goal = Goal.new()
+    # How is it going to know its the exact goal you're trying to submit??
+    # need to give it a date from the goal itself
+    submission = Submission.new(params[:goal_id])
+    @date = Goal.find(params["goal_id"].date_id)
+    @goal = Goal.find(params["goal_id"])
+    submission.goal = @goal
+    submission.user = current_user
+    submission.save
+    raise
+  end
+
+  # Copy paste this and delete this before pushing (remember you're still on the created-submission-forms-branch)
+  def create
+    submission = Submission.new(submission_params) # needs to be getting date through some params i think -> Ideally want it to be getting it through the seeded data. Submission.start_date????
+
+    # uncomment here later
+    @goal = Goal.find(params["goal_id"])
+    submission.goal= @goal
+    submission.user = current_user
+    if submission.save
+      # where do I want to redirect to? home path??
+      redirect_to home_path()
```

Comment:
Not sure, but maybe the home_path doesn't need the ()

Reply:
I think it doesn't matter. It works either way